### PR TITLE
package release prep

### LINKED ISSUE #114: Release preparation for `cell_feature_data` package
## Use Case

Ensure that all necessary preparations are completed for the release of the package, so that it is ready for publishing on PyPI. 

## Acceptance Criteria

_Please describe how you know this is done_

- [x] MIT license is updated #113 
- [x] Unit tests have been written and passed 
- [x] add CI workflow #112

The packaged is published to PyPI 
- [x] verify the distribution 
- [x] verify release
- [x] get PyPI credentials 
- [x] test the release

- [x] Readme is updated

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
Partially closes #114 

- [x] Awaiting PYPI publish 

Estimate review time: 10-15mins medium  

Solution
========
What I/we did to solve this problem
- added unit tests, `pyproject.toml`, `LICENSE`, `.gitignore` and continuous integration workflow
- updated docstrings and README

~~Looking for input~~: the python package `cell_feature_data` is currently within the `cell-feature-data` javascript codebase. I encountered the `ModuleNotFoundError: No module named 'cell_feature_data'` a few times due to directory structure issues. I resolved this by adding `PYTHONPATH` to the virtual environment's activation script and cd to the correct directory in the workflow. I'm concerned this issue may affect other users as well. I'd appreciate any testing or feedback on how best to handle it. **We are using a double-directory structure to make sure that `setuptools` can identify the root directory correctly**

After merged into main, we can create more badges for the package.
 
## Type of change
Please delete options that are not relevant.
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Follow the [instruction](https://github.com/allen-cell-animated/cell-feature-data/tree/feature/release-prep?tab=readme-ov-file#option-1-use-the-cell_feature_data-package) in the updated README

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  test:
+    if: ${{ !contains(github.event.head_commit.message, 'Bump version') }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [3.9]
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+
+    steps:
+      - uses: actions/checkout@v4
```
Comment (ShrimpCryptid) [inline .github/workflows/ci.yml:32]: I believe software recommends we use actions tagged to a specific commit hash and not a version number?

File: cell_feature_data/tests/test_data_loader.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+def test_format_display_name():
+    assert (
+        FeatureDefsDoc.format_display_name("interphase-or-mitosis")
+        == "Interphase or Mitosis"
+    )
+    assert (
+        not FeatureDefsDoc.format_display_name("interphase-or-mitosis")
+        == "Interphase Or Mitosis"
+    )
```
Comment (ShrimpCryptid) [inline cell_feature_data/tests/test_data_loader.py:78]: Should you test the other words that aren't capitalized? Or test what happens when they're at the beginning of the string?

**Review by ShrimpCryptid — APPROVED** [review]
Couple small comments/suggestions but otherwise looks fine 👍

File: README.md [hunk 3]
```diff
@@ -3,16 +3,17 @@
...
 #### Option 1: Use the `cell_feature_data` package
-1. Make a branch or fork of this repo
-2. Create a virtual environment if not already created: `python3 -m venv venv/[ENV-NAME]`
-3. Activate the virtual environment: `source [ENV-PATH]/bin/activate`
-4. Install the dependencies: `pip install -e .` This step also makes the `create-dataset` command available globally within the virtual environment.
-5. Run `create-dataset` to start the dataset creation process. This will: 
+1. Create a virtual environment if not already created: `python3 -m venv venv/[ENV-NAME]`
+2. Activate the virtual environment: 
+   - On macOS/Linux: `source [ENV-PATH]/bin/activate`
+   - On Windows: `venv/[ENV-PATH]\Scripts\activate`
```
[the remark this answers is not in the document]
Comment (rugeli) [reply README.md]: Thank you! The `venv/`is indeed a typo, fixed!

File: README.md [hunk 4]
```diff
@@ -3,16 +3,17 @@
 # For dataset creators:
 ## To create a new dataset:
 #### Option 1: Use the `cell_feature_data` package
-1. Make a branch or fork of this repo
-2. Create a virtual environment if not already created: `python3 -m venv venv/[ENV-NAME]`
-3. Activate the virtual environment: `source [ENV-PATH]/bin/activate`
-4. Install the dependencies: `pip install -e .` This step also makes the `create-dataset` command available globally within the virtual environment.
-5. Run `create-dataset` to start the dataset creation process. This will: 
+1. Create a virtual environment if not already created: `python3 -m venv venv/[ENV-NAME]`
```
Comment (toloudis) [inline README.md:6]:
```suggestion
1. Create a virtual environment if not already created: `python3 -m venv [ENV-PATH]`
```
Suggesting this to match the following step 2.  What do you think?

Comment (rugeli) [reply README.md:6]: ah good catch! Fixed!

**Comment by rugeli** [comment]
next steps: 
- verify the package name with @djeung. (I have it as `cell_feature_data` in the pyproject file but it appears as `cell-feature-data`)
Screenshot 2024-09-24 at 5 17 34 PM

- Create a separate README for the package and update the existing README

File: cell_feature_data/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Cell Feature Data
+
+[![Dataset validation](https://github.com/allen-cell-animated/cell-feature-data/actions/workflows/validate.yml/badge.svg?branch=main)](https://github.com/allen-cell-animated/cell-feature-data/actions/workflows/validate.yml)
```
Comment (rugeli) [inline cell_feature_data/README.md:3]: more badges(e.g. license, build, version, etc) will be added once merged to `main`

**Comment by rugeli** [comment]
package on PyPI: https://pypi.org/project/cell-feature-data/

**Review by toloudis — APPROVED** [review]
LGTM!  I am looking forward to taking some time soon to actually try it hands-on.  I think that will help inform next steps (if any).

**Pull request merged**